src/kaltura-live-plugin.tsx

```diff
@@ -109,16 +105,24 @@ export class KalturaLivePlugin
     onMediaUnload(): void {
         this.isMediaLive = false;
         this.isLiveMediaLoaded = false;
+    }
+
+    onPluginDestroy(): void {
         this._resetTimeout();
         this._player.removeEventListener(this._player.Event.FIRST_PLAY, this._handleFirstPlay);
-
         this._player.removeEventListener(this._player.Event.TIMED_METADATA, this.handleTimedMetadata);
-        this._player.removeEventListener(this._player.Event.ENDED, this._handleEnd);
-        this._player.removeEventListener(this._player.Event.ABORT, this._handleEnd);
+        this._player.removeEventListener(this._player.Event.ENDED, this._handleEndOrAbort);
+        this._player.removeEventListener(this._player.Event.ABORT, this._handleEndOrAbort);
     }
 
     private _activatePlugin = () => {
```

Comment:
It won't work for change media so:
The most correct way is to add `loadMedia` which declared on CorePlugin in Contrib, adding the logic of _activatePlugin to `loadMedia` method and set the logic of onPluginDestroy to onMediaUnload, then events will be registered and remove in the same flow of media. 
I suggest adding eventManager instead of managing events separately - it exists on core.